search_agg_order.go

```diff
@@ -0,0 +1,30 @@
+package opensearchtools
+
+import (
+	"fmt"
+)
+
+// Order encapsulates the sorting capabilities for [Aggregation] requests to OpenSearch.
+// An empty Order will be rejected by OpenSearch as the target must be non-null and non-empty
+type Order struct {
+	Target string
+	Desc   bool
+}
+
+// NewOrder instantiates an aggregation Order with the target and whether it should be descending or ascending.
+func NewOrder(field string, desc bool) Order {
+	return Order{
+		Target: field,
+		Desc:   desc,
+	}
+}
+
+// ToOpenSearchJSON converts the Order to the correct OpenSearch JSON.
+func (o Order) ToOpenSearchJSON() ([]byte, error) {
+	dir := "asc"
+	if o.Desc {
+		dir = "desc"
+	}
+
+	return []byte(fmt.Sprintf("{%q: %q}", o.Target, dir)), nil
```

Comment:
Maybe add a check here that `o.Target` is not empty and return an explicit error rather than generating and returning invalid JSON.